comment_scan/replier.rb

```diff
@@ -0,0 +1,296 @@
+    comment = MessageCollection::ALL_ROOMS.comment_for(parent_id.to_i)
+    return if comment.nil?
+
+    replier.chatter.say("Currently marked #{comment.tps.to_i}tps/#{comment.fps.to_i}fps", room_id)
+end
+
+def del_reply(replier, msg_id, parent_id, room_id, *args)
+    comment = MessageCollection::ALL_ROOMS.comment_for(parent_id.to_i)
+    return if comment.nil?
+
+    MessageCollection::ALL_ROOMS.message_ids_for(comment).each do |id|
+        replier.chatter.delete(id)
+    end
+end
+
+def huh(replier, msg_id, parent_id, room_id, *args)
+    comment = MessageCollection::ALL_ROOMS.comment_for(parent_id.to_i)
+    return if comment.nil?
+
+    matched_regexes = replier.scanner.report_raw(comment["post_type"], comment["body_markdown"])
+    # Go through regexes we matched to build reason_text
+    reason_text = matched_regexes.map do |regex_match|
+        reason = "Matched reason \"#{regex_match.reason.name}\""
+        regex = "for regex #{regex_match.regex}"
+        "#{reason} #{regex}"
+    end.join("\n")
+
+    # If post isn't deleted, check if this was an inactive comment
+    if post = replier.seclient.post_exists?(comment.post_id)
+        if replier.scanner.timestamp_to_date(post.json["last_activity_date"]) < replier.scanner.timestamp_to_date(comment["creation_date"]) - 30
+            reason_text += "Comment was made #{(replier.scanner.timestamp_to_date(comment["creation_date"]) - replier.scanner.timestamp_to_date(post.json["last_activity_date"])).to_i} days after last activity on post\n"
+        end
+    end
+
+    reason_text += "\nComment has toxicity of #{comment["perspective_score"]}" if comment["perspective_score"].to_f >= 0.7
+
+    replier.chatter.say((reason_text.empty? ? "Comment didn't match any regexes" : reason_text), room_id)
+end
+
+def rescan(replier, msg_id, parent_id, room_id, *args)
+    db_comment = MessageCollection::ALL_ROOMS.comment_for(parent_id.to_i)
+    return if db_comment.nil?
+
+    if replier.seclient.comment_deleted?(db_comment["comment_id"])
+        replier.chatter.say("Comment with id #{db_comment["comment_id"]} was deleted and cannot be rescanned.", room_id)
+    else
+        replier.scanner.scan_comments(db_comment["comment_id"])
+    end
+end
+
+def report(replier, msg_id, parent_id, room_id, *report_reason)
+    db_comment = MessageCollection::ALL_ROOMS.comment_for(parent_id.to_i)
+    return if db_comment.nil?
+
+    if replier.seclient.comment_deleted?(db_comment["comment_id"])
+        replier.chatter.say("Comment with id #{db_comment["comment_id"]} was deleted and cannot be reported.", room_id)
+    else
+        replier.scanner.custom_report(db_comment, "Reported with custom reason: #{report_reason.join(' ')}")
+    end
+end
+
+def howgood_tp(replier, msg_id, parent_id, room_id, num_to_display=3)
+    hg_comment = MessageCollection::ALL_ROOMS.howgood_for(parent_id.to_i)
+    return if hg_comment.nil?
+
+    if !/\A\d+\z/.match(num_to_display) || num_to_display.to_i < 1
```

Comment:
What's this regex supposed to check? For "is this string an int" I generally prefer `str.to_i.to_s == str`

Reply:
Ooh `to_i` doesn't throw when it fails? I still need to get used to working in something that's not as mean as C++. That regex just checks that all we're given is some digits, but I like yours.